hydra-node/src/Hydra/Chain/Direct/Wallet.hs

```diff
@@ -586,19 +587,22 @@ stateQueryClient tracer tipVar utxoVar address =
       check (tip == genesisPoint)
     pure $ LSQ.SendMsgReAcquire Nothing clientStAcquiring
 
-  handleEraMismatch :: m (LSQ.ClientStAcquired Block (Point Block) (Query Block) m ())
-  handleEraMismatch = do
-    -- FIXME: log something before looping back?
+  handleEraMismatch :: MismatchEraInfo (CardanoEras StandardCrypto) -> m (LSQ.ClientStAcquired Block (Point Block) (Query Block) m ())
+  handleEraMismatch (mkEraMismatch -> Ouroboros.EraMismatch{Ouroboros.ledgerEraName, Ouroboros.otherEraName}) = do
+    traceWith tracer $ EraMismatchError ledgerEraName otherEraName
     threadDelay 30
     reset
 
 --
 -- Logs
 --
 
+-- NOTE(AB): We don't use records here because there's no ToJSON insetance for SomePoint so there's no point
+-- in using a Generic ToJSON instance
```

Comment:
Typo `insetance` and not sure what this is explaining. Why there are no field names for the two `Text` below?

Reply:
No other constructors has field names, adding them only here would feel quite odd. Adding them everywhere would make more sense, but then they are never used and could not be used in the ToJSON instance, hence my reluctance to add them. I am fine adding fields if that makes more sense to you.